src/App.tsx

```diff
@@ -1,14 +1,50 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos, getUser } from './api';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [loadingUser, setLoadingUser] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [user, setUser] = useState<User | null>(null);
+  const [filter, setFilter] = useState<'all' | 'completed' | 'active'>('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    const feychTodos = async () => {
+      const todosData = await getTodos();
+
+      setTodos(todosData);
+      setLoading(false);
+    };
+
+    feychTodos();
+  }, []);
+
+  const openModal = async (todo: Todo) => {
+    setSelectedTodo(todo);
+    setLoadingUser(true);
+    const userData = await getUser(todo.userId);
+
+    setUser(userData);
+    setLoadingUser(false);
```

Comment:
Similarly, add error handling for the `getUser` function call to manage any errors that might occur while fetching user data.